### ISSUE TITLE
ingress_with_source_security_group_id not working with created SG

### ISSUE THREAD

**Issue by sokecillo** [terraform-aws-modules/terraform-aws-security-group#41]
When you create a SG and then you want to use this SG on another SG ingress_with_source_security_group_id rule, it fails with the following error: 

```
* module.test-security-group.aws_security_group_rule.ingress_with_source_security_group_id: 
aws_security_group_rule.ingress_with_source_security_group_id: value of 'count' cannot be computed
```

Sample Code: 

```
module "bastion-ec2-security-group" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "1.15.0"
  name    = "TestEC2SG"
  vpc_id  = "vpc-xxxxxxx"

  egress_with_cidr_blocks = [
    {
      from_port   = 0
      to_port     = 65535
      protocol    = "-1"
      cidr_blocks = "0.0.0.0/0"
    },
  ]
}

module "bastion-security-group" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "1.15.0"
  name    = "TestSG"
  vpc_id  = "vpc-xxxxxxx"

  ingress_with_source_security_group_id = [
    {
      from_port   = 22
      to_port     = 22
      protocol    = "tcp"
      description = "SSH"
      source_security_group_id = "${module.bastion-ec2-security-group.this_security_group_id}"
    },
  ]
  egress_with_cidr_blocks = [
    {
      from_port   = 0
      to_port     = 65535
      protocol    = "-1"
      cidr_blocks = "0.0.0.0/0"
    },
  ]
}
```

**Comment by antonbabenko** [comment]
Unfortunately, this is a known limitation of Terraform and this module. Read notes [in usage section](https://github.com/terraform-aws-modules/terraform-aws-security-group#usage).

**Comment by antonbabenko** [comment]
This is now fixed in [v2.0.0](https://github.com/terraform-aws-modules/terraform-aws-security-group/releases/v2.0.0). See this section in [README](https://github.com/terraform-aws-modules/terraform-aws-security-group#note-about-value-of-count-cannot-be-computed)
